docs/features/wait/walk.md

```diff
@@ -2,11 +2,18 @@
 
 Walk walks the strategies tree and calls the visit function for each node.
 
+This allows modules to easily amend default wait strategies, updating or
+removing specific strategies based on requirements of functional options.
+
+For example removing a TLS strategy if a functional option enabled insecure mode
+or changing the location of the certificate based on the configured user.
+
```

Comment:
Fantastic! Thanks for making it clear 🙏

Reply:
Sorry you said TLS specifically, this is a FileStrategy removal example, we can add a TLS one when it's integrated into couchbasedb as it will be used there.

Reply:
Indeed, just mentioned the above example for 
> For example removing a TLS strategy if a functional option enabled insecure mode
or changing the location of the certificate based on the configured user.

Fine for me if we elaborate on this as a follow-up 👍